AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOrchestrator.swift

```diff
@@ -180,3 +188,38 @@ extension AWSInitialSyncOrchestrator: Resettable {
         onComplete()
     }
 }
+
+@available(iOS 13.0, *)
+extension AWSInitialSyncOrchestrator {
+    private typealias ResponseType = PaginatedList<AnyModel>
+    private func graphqlErrors(from error: GraphQLResponseError<ResponseType>?) -> [GraphQLError]? {
+        if case let .error(errors) = error {
+            return errors
+        }
+        return nil
+    }
+
+    private func errorTypeValueFrom(graphQLError: GraphQLError) -> String? {
+        guard case let .string(errorTypeValue) = graphQLError.extensions?["errorType"] else {
+            return nil
+        }
+        return errorTypeValue
+    }
+
+    private func isUnauthorizedError(_ error: DataStoreError) -> Bool {
+        guard case let .sync(_, _, underlyingError) = error,
+              let datastoreError = underlyingError as? DataStoreError
+              else {
+            return false
+        }
+
+        if case let .api(apiError, _) = datastoreError,
+           let responseError = apiError as? GraphQLResponseError<ResponseType>,
+           let graphQLError = graphqlErrors(from: responseError)?.first,
+           let errorTypeValue = errorTypeValueFrom(graphQLError: graphQLError),
+           case .unauthorized = AppSyncErrorType(errorTypeValue) {
+            return true
+        }
+        return false
+    }
+}
```

Comment:
This is logic is a copy of what's in `AWSModelReconciliationQueue`. It should be moved to an extension of one of the errors (.i.e. `DataStoreError`) or to a shareable static method